### PR TITLE
[IMP] payment: better SCA flow for stripe

### PR DESCRIPTION
**Pull request by bouvyd**

### CODE REVIEW

File: addons/payment_stripe/static/src/js/payment_form.js [hunk 1]
```diff
@@ -162,6 +221,8 @@ PaymentForm.include({
         // first we check that the user has selected a stripe as s2s payment method
         if ($checkedRadio.length === 1 && this.isNewPaymentRadio($checkedRadio) && $checkedRadio.data('provider') === 'stripe') {
             return this._createStripeToken(ev, $checkedRadio);
+        } else if ($checkedRadio.attr('name') === 'pm_id' && !this.isNewPaymentRadio($checkedRadio) && !this.isFormPaymentRadio($checkedRadio) && $checkedRadio.data('provider') === 'stripe') {
```
Comment (nla-odoo) [inline addons/payment_stripe/static/src/js/payment_form.js]: I think we should merge this with 1st condition
`($checkedRadio.length === 1 && (this.isNewPaymentRadio($checkedRadio) || ($checkedRadio.attr('name') === 'pm_id') && $checkedRadio.data('provider') === 'stripe')`

File: addons/payment_stripe/static/src/js/payment_form.js [hunk 2]
```diff
@@ -75,9 +76,67 @@ PaymentForm.include({
...
+            self.enableButton(button);
+            self.displayError(
+                _t('Unable to save card'),
+                _t("We are not able to add your payment method at the moment. ") +
+                    self._parseError(error)
+            );
+        });
+    },
+    _chargeStripeToken: function (ev, $checkedRadio) {
+        var self = this;
```
Comment (nla-odoo) [inline addons/payment_stripe/static/src/js/payment_form.js]: Is it possible to create a separate method for creating charge as most of the code repeates in _chargeStripeToken and _createStripeToken??

Comment (dbo-odoo) [reply addons/payment_stripe/static/src/js/payment_form.js]: good point i'll factorize the common code in a private method

Comment (dbo-odoo) [reply addons/payment_stripe/static/src/js/payment_form.js]: done!

File: addons/payment_stripe/static/src/js/payment_form.js [hunk 3]
```diff
@@ -88,6 +90,49 @@ PaymentForm.include({
...
+                _t("We were not able to process your payment. ") +
+                    self._parseError(error)
+            );
+        });
+    },
+    _chargeStripeToken: function(stripe, json_tx_url, formData) {
+        var tx_reference;
+        var self = this;
+        return this._rpc({
+            route: json_tx_url,
```
Comment (nla-odoo) [inline addons/payment_stripe/static/src/js/payment_form.js]: we can get json_tx_url from this.$el.data('json-action') directly, so I don't think we should pass it from both the place.

File: addons/payment_stripe/static/src/js/payment_form.js [hunk 4]
```diff
@@ -49,6 +49,7 @@ PaymentForm.include({
         var formData = self.getFormData(inputsForm);
         var stripe = this.stripe;
         var card = this.stripe_card_element;
+        var tx_reference;
```
Comment (nla-odoo) [inline addons/payment_stripe/static/src/js/payment_form.js]: tx_reference is also not used

File: addons/payment_stripe/static/src/js/payment_form.js (hunk 3 above)
Comment (dbo-odoo) [reply addons/payment_stripe/static/src/js/payment_form.js]: agreed

**Comment by dbo-odoo** [comment]
superseeded by #37621

**Pull request closed without merging**